### PR TITLE
feat(gateway): make request body size configurable via env

### LINKED ISSUE #116: Make Request Body Size Configurable
## Problem

The Gateway currently enforces a hardcoded request body size limit of **10MB** in `gateway/main.go`.

```go
// gateway/main.go line ~363
const maxBodySize = 10 * 1024 * 1024
```

Hardcoded limits are an anti-pattern for production services. Different deployments may need to handle larger documents for summarization or, conversely, restrict payloads to very small sizes to prevent DoS attacks on memory.

## Solution

Refactor this constant into an environment variable `MAX_REQUEST_BODY_MB` with a default value of 10.

## Implementation

1.  **Add Helper**: Add a configuration helper `getMaxBodySize()` in `gateway/config.go`.
    -   Read `MAX_REQUEST_BODY_MB` (int).
    -   Default to `10`.
    -   Return bytes (`val * 1024 * 1024`).
2.  **Update Handler**: Replace the `const maxBodySize` in `handleSummarize` with the dynamic value from the helper.
3.  **Update Documentation**: Add the new variable to `.env.example`.

### Code Hint

```go
// gateway/config.go
func getMaxBodySize() int64 {
    mb := getEnvAsInt("MAX_REQUEST_BODY_MB", 10)
    return int64(mb) * 1024 * 1024
}

// gateway/main.go
func handleSummarize(c *gin.Context) {
    // ...
    maxSize := getMaxBodySize()
    c.Request.Body = http.MaxBytesReader(c.Writer, c.Request.Body, maxSize)
    // ...
}
```

## Acceptance Criteria

- [ ] `MAX_REQUEST_BODY_MB` environment variable controls the limit.
- [ ] Default remains 10MB if variable is unset.
- [ ] `.env.example` includes the new variable.

## Testing

```bash
# 1. Set MAX_REQUEST_BODY_MB=1 in .env
# 2. Restart Gateway
# 3. Send a 2MB payload
curl -X POST localhost:3000/api/ai/summarize -d @large_file.json
# Expect: 413 Request Entity Too Large
```

### PR DESCRIPTION
**Pull request by PranavAgarkar07**
## Summary

Replaces the hardcoded `const maxBodySize = 10 * 1024 * 1024` in `gateway/main.go` with a configurable value driven by the `MAX_REQUEST_BODY_MB` environment variable. A new helper `getMaxBodySize()` in `gateway/config.go` handles parsing and falls back to 10 MB when the variable is unset or invalid.

Also fixes a P1 issue introduced when the original hardcoded constant was replaced: the 413 error response JSON key drifted from `"max_size"` to `"max_size_mb"` with a type change (string → int). Restored the original response format so clients parsing the error do not break.

## Related Issue

Closes #116

## Type Of Change

- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Tests
- [ ] Refactor
- [ ] Deployment/config

## Affected Areas

- [x] Gateway (`gateway/`)
- [ ] Verifier (`verifier/`)
- [ ] Web (`web/`)
- [ ] E2E/tests
- [ ] Benchmarks
- [ ] Deployment/config
- [ ] Documentation/community files

## Contributor Checklist

- [x] I kept the change focused and avoided unrelated refactors.
- [x] I updated README/service docs/OpenAPI/env examples when behavior, config, headers, status codes, or public APIs changed.
- [x] I did not commit secrets, private keys, funded wallets, API keys, or real production URLs.
- [x] I checked x402/EIP-712 field parity when touching payment context, signatures, timestamps, nonces, chain IDs, receipts, or wallet flow.
- [x] I checked Docker/Compose/Fly/Vercel docs when touching ports, service names, health checks, or environment variables.

## Verification

```text
$ cd gateway && go vet ./... && go test -v ./...
ok  gateway	10.009s
ok  gateway/internal/ai	0.006s
```

## Notes For Reviewers

The `getMaxBodySize()` helper lives in `config.go` alongside the other typed-helper functions (`getEnvAsDuration`, `getEnvAsBool`, etc.) and follows the same pattern: log a warning and return the default when parsing fails.

## Summary by CodeRabbit

* **New Features**
  * Request body size limit is now configurable via environment variable (default 10 MB, maximum 100 MB). Oversized requests return a 413 with the configured limit shown.

* **Documentation**
  * Added docs and example env entries for the new request body size setting.

* **Tests**
  * Added tests covering configuration parsing, clamping behavior, and enforcement of the request-body limit.

### CODE REVIEW

**Review by AnkanMisra — CHANGES_REQUESTED** [review]
Resolve conflicts

**Comment by PranavAgarkar07** [comment]
Resolved merge conflicts with main (web/package.json scripts moved to bun:test, web/src/lib/errors.test.ts import changed from vitest to bun:test).

Also addressed the two CodeRabbit findings:

**Finding 1 (config.go):** Added non-positive guard to getMaxBodySize — treats MAX_REQUEST_BODY_MB=0 or negative as the default 10MB with a logged warning, matching getPositiveTimeout's pattern.

**Finding 2 (cache.go):** Replaced hardcoded const maxBodySize = 10*1024*1024 with getMaxBodySize() and dynamic fmt.Sprintf("%dMB", maxSize/1024/1024) for the 413 response. CACHE_ENABLED now honors MAX_REQUEST_BODY_MB.

**Comment by AnkanMisra** [comment]
vercel logs:
```
19:11:23.010 Running build in Washington, D.C., USA (East) – iad1
19:11:23.011 Build machine configuration: 2 cores, 8 GB
19:11:23.221 Cloning github.com/AnkanMisra/MicroAI-Paygate (Branch: feat/configurable-body-size, Commit: 2538326)
19:11:24.355 Cloning completed: 1.134s
19:11:24.845 Restored build cache from previous deployment (3ynGXicmuDhWAVvP5YkpEPmJqbVS)
19:11:25.070 Running "vercel build"
19:11:25.091 Vercel CLI 54.7.1
19:11:25.319 Running "install" command: `bun install --frozen-lockfile`...
19:11:25.359 bun install v1.3.12 (700fc117)
19:11:25.383 Resolving dependencies
19:11:26.606 Resolved, downloaded and extracted [268]
19:11:26.608 error: lockfile had changes, but lockfile is frozen
19:11:26.608 note: try re-running without --frozen-lockfile and commit the updated lockfile
19:11:26.622 Error: Command "bun install --frozen-lockfile" exited with 1
```

**Comment by AnkanMisra** [comment]
@codex review

**Review by AnkanMisra — CHANGES_REQUESTED** [review]
[Review](https://github.com/AnkanMisra/MicroAI-Paygate/pull/186#pullrequestreview-4419171344)

**Comment by PranavAgarkar07** [comment]
Addressed all 4 Codex findings in a0484b1:

**P2 — Remove unused Vitest dependency:** Removed vitest from web/package.json devDependencies and regenerated web/bun.lock. The test:unit script uses bun test, not vitest.

**P2 — Document MAX_REQUEST_BODY_MB:** Added to gateway/README.md optional-variable table (default: 10, clamped to 10240) and .env.production.example as a commented-out entry.

**P2 — Add regression coverage:** Added TestGetMaxBodySize with 6 sub-tests covering: default (unset), custom value (25MB), zero fallback, negative fallback, overflow clamp (99999→10240), and non-numeric fallback.

**P3 — Clamp MAX_REQUEST_BODY_MB before multiplying:** Added maxBodySizeMBMax = 10240 constant and clamp in getMaxBodySize(). Values exceeding 10240 MB are logged and clamped to prevent int64 overflow.

**Comment by AnkanMisra** [comment]
@codex review

**Comment by AnkanMisra** [comment]
@PranavAgarkar07 which issue does it solve, please mention it in the description

**Comment by PranavAgarkar07** [comment]
Done — added `Closes #116` in the PR description under a new "Related Issue" section.

**Comment by PranavAgarkar07** [comment]
Addressed both open Codex findings from the second review (commit `a0484b1`):

**P2 — Cap configurable request bodies below memory-exhaustion size**
- Reduced `maxBodySizeMBMax` from 10240 MB (10 GiB) to **100 MB** — still 10x the default, but prevents a single request from forcing the gateway to buffer 10 GiB in memory
- Updated docs and tests to match

**P2 — Revert unrelated web lockfile updates**
- Reverted `web/bun.lock` to `origin/main` version, dropping 501 lines of unrelated transitive dependency bumps (Babel, TypeScript ESLint packages, etc.)

Ran `go test -v ./...` and `go vet ./...` — all passing.

**Comment by AnkanMisra** [comment]
@codex review the pr

File: .env.example [hunk 1]
```diff
@@ -75,6 +75,10 @@ RATE_LIMIT_VERIFIED_RPM=120
 # Cleanup interval for stale buckets (seconds)
 RATE_LIMIT_CLEANUP_INTERVAL=300
 
+# Request Body Size Configuration
+# Maximum request body size in MB (default: 10)
+MAX_REQUEST_BODY_MB=10
```
[the remark this answers is not in the document]
Comment (AnkanMisra) [reply .env.example:80]: @PranavAgarkar07

File: gateway/config_test.go [hunk 2]
```diff
@@ -474,3 +474,47 @@ func TestGetReceiptTTL(t *testing.T) {
...
 }
+
+func TestGetMaxBodySize(t *testing.T) {
+	t.Run("default when unset", func(t *testing.T) {
+		t.Setenv("MAX_REQUEST_BODY_MB", "")
+		if got := getMaxBodySize(); got != 10*1024*1024 {
+			t.Fatalf("expected 10MB default, got %d", got)
+		}
+	})
+
```
Comment (AnkanMisra) [inline gateway/config_test.go:494]: ### Add route-level regression coverage for configured 413 responses

These added tests only validate `getMaxBodySize()` parsing, but they never exercise `handleSummarize` or `CacheMiddleware` with `MAX_REQUEST_BODY_MB` set to a smaller value and an oversized request body

That leaves both the cached and **uncached 413** responses unprotected, so a future drift in the actual rejection path or max_size payload would still pass CI even though this PR changes a public error contract there

**Review by AnkanMisra — CHANGES_REQUESTED** [review]
@PranavAgarkar07 Final Changes wrt to pr

**Comment by PranavAgarkar07** [comment]
Addressed review feedback - added TestRequestBodyLimit_EnforcedAtRouteLevel with 5 sub-tests covering body size enforcement at the route level (below/at/over limit with env-driven config). Debug test removed. Commit 66adb49. All tests pass.
